test/e2e/config/e2e_conf.yaml

```diff
@@ -111,7 +111,7 @@ variables:
   KUBERNETES_VERSION: "v1.25.2"
   # INIT_WITH_KUBERNETES_VERSION will be used here
   # https://github.com/kubernetes-sigs/cluster-api/blob/bb377163f141d69b7a61479756ee96891f6670bd/test/e2e/clusterctl_upgrade.go#L170
-  INIT_WITH_KUBERNETES_VERSION: ${KUBERNETES_VERSION}
+  INIT_WITH_KUBERNETES_VERSION: "v1.23.8"
```

Comment:
Please add a comment explaining that this var used by management cluster upgrade and it is fixed to 1.23.8 because v1a4 capi does not support k8v version higher